test/Transforms/add-outer-parallel-loop.mlir

```diff
@@ -0,0 +1,22 @@
+// RUN: imex-opt --imex-add-outer-parallel-loop %s | FileCheck %s
+func.func @test(%arg0: memref<10x20xf32>) -> memref<f32> {
+  %c0 = arith.constant 0 : index
+  %c10 = arith.constant 10 : index
+  %c1 = arith.constant 1 : index
+  %c20 = arith.constant 20 : index
+  %cst = arith.constant 0.000000e+00 : f32
+  %alloc = memref.alloc() {alignment = 128 : i64} : memref<f32>
+  memref.store %cst, %alloc[] : memref<f32>
+  %alloc_0 = memref.alloc() {alignment = 128 : i64} : memref<f32>
+  memref.copy %alloc, %alloc_0 : memref<f32> to memref<f32>
+  // CHECK: scf.parallel
```

Comment:
please check for loops are actually inside this `scf.parallel`

Reply:
added

Reply:
nit: you can improve this check by also matching closing `}`